Programação R2.1-2.25

GuiCT opened the pull request:
Classes:
- [x] ControladorProcesso
- [x] Processo
- [x] Anexo
- [x] Petição
- [x] Minuta

GuiCT committed:
feat: incluindo classe Anexo e suas subclasses

GuiCT committed:
fix: corrigindo lógica de editarMinuta

GuiCT committed:
fix: tornando criarPeticao em método estático

GuiCT committed:
fix: tornando criarMinuta em método estático

GuiCT committed:
feat: criando classe "Processo"

GuiCT committed:
fix: utilizando serialização para guardar estado

abandonando uso de Database pelo overhead

GuiCT committed:
feat: várias mudanças

Melhorando estruturação da GUI
Serializando a instância da classe FCToga
Utilizando injeção de dependências para gerar GUI

GuiCT committed:
feat: caso de uso de visualizar histórico de proc.

criando GUI e lógica de passagem de dados

GuiCT committed:
feat: caso de uso de alterar minuta

GuiCT committed:
Atualizando Tabela de Anexos (tempo real)

GuiCT committed:
feat: caso de uso de assinar minuta

por enquanto sem tratamento de nome de juiz e comarca

GuiCT committed:
feat: caso de uso de adicionar anexo

GuiCT committed:
feat: caso de uso de visualizar um anexo

Transformando FCToga em uma instância singleton

GuiCT committed:
fix: commitando arquivo de GUI para visualização

GuiCT committed:
melhorando legibilidade e estruturação

Separando classes da view
Utilizando constantes para facilitar a montagem da GUI
Tornando FCToga uma instância singleton
Desinchando Controlador de Processo

GuiCT committed:
corrigindo chamada a método da GUI

GuiCT commented:
Closes #7

becelli reviewed (changes requested):
# SQA da programação

Início 26/01/2023 - Fim 26/01/2023

## Programação

| Defeito | Tipo | Descrição                                                                                      |
| ------- | ---- | ---------------------------------------------------------------------------------------------- |
| 01      | FI   | O campo de CPF no cadastro não está aceitando CPFs que iniciam-se com 0.                       |
| 03      | O    | As minutas não estão sendo persistidas.                                                        |
| 04      | O    | As petições não estão sendo persistidas.                                                       |
| 05      | O    | Não está sendo possível visualizar minutas.                                                    |
| 06      | FI   | R2.4, R2.9 e R2.10 - O representante não está sendo incluído.                                  |
| 07      | FI   | R2.12 - A classificação de uma minuta não está sendo limitada a despacho, decisão ou sentença. |
| 08      | FI   | R2.6 e R2.7 - O processo não está sendo iniciado com uma petição                               |
| 09      | FI   | R2.8 - O tipo de processo não está sendo limitado a civil ou criminal.                         |
| 10      | FI   | R2.11 - O nome do processo não está de acordo com o Apêndice C do documento de requisitos.     |
| 11      | O    | R2.19-2.25 - O sistema não está exibindo o histórico do processo.                              |

GuiCT committed:
fix: corrigindo defeitos de persistência e de vis.

GuiCT committed:
fix: permitindo CPF iniciado em 0.

Utilizando InputVerifier para impedir que o usuário altere o foco sem cumprir com o regex.

GuiCT committed:
fix: limitando tipos de Minuta

GuiCT committed:
fix: limitando tipos de cadastro

Na primeira execução, pode ser cadastrado apenas um Administrador.

GuiCT committed:
fix: incluindo petição inicial, limitando tipos

Processo agora só pode ser Criminal ou Civil
Justiça Pública é automaticamente requerente de todo processo criminal
Corrigindo número de processo.

GuiCT committed:
fix: tabela agora cita CNPJ ao invés de apenas CPF

GuiCT committed:
feat: testando novos layouts de GUI

GuiCT committed:
feat: exibindo histórico do proceso de forma correta

Exibindo número, data de criação, data de última alteração, etc.

GuiCT committed:
feat: incluindo InputVerifiers

Salies commented:
Merge provisório para continuação da programação no Sprint 3.
O SQAs final das Sprints 2 e 3 serão feitos juntos.
